src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/WaySequence.java

```diff
@@ -0,0 +1,84 @@
+package org.openstreetmap.josm.plugins.pt_assistant.data;
+
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.data.osm.RelationMember;
+import org.openstreetmap.josm.data.osm.Way;
+
+import static org.openstreetmap.josm.plugins.pt_assistant.utils.RouteUtils.isPTWay;
+
+public final class WaySequence<previousWay extends Way, currentWay extends Way, nextWay extends Way, wayAfterNextWay extends Way> {
```

Comment:
@PolyglotOpenstreetmap You can remove the angle brackets, the way you use them here doesn't do anything.

If you write `WaySequence<T extends Way>` you are defining a generic type T, that can be a `Way` or any class extending `Way` (or implementing `Way` if `Way` was an interface).

The convention is to use a single uppercase letter, typically `T`, if it's the only generic type.

The generic type can then be used throughout the file just as you would use the type `Way`:
```java
…
public T currentWay;
…
public WaySequence(final T previousWay, final T currentWay, final T nextWay) {
}
…
```
But I think at the moment you don't need generics here.

Reply:
Thanks, I changed this. Now the code looks a lot cleaner.